src/Sulu/Component/Webspace/Manager/WebspaceCollectionBuilder.php

```diff
@@ -121,45 +122,14 @@ public function build()
                 $this->webspaces[] = $webspace;
 
                 $this->buildPortals($webspace);
-            } catch (\InvalidArgumentException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (InvalidUrlDefinitionException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (InvalidCustomUrlException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
-            } catch (WebspaceLocalizationNotUsedException $e) {
-                $this->logger->warning(
-                    sprintf(
-                        'Error: "%s" in "%s". The file has been skipped.',
-                        $e->getMessage(),
-                        $file->getRealPath()
-                    )
-                );
+            } catch (\Exception $e) {
```

Comment:
You are in a for-each loop here. An error could happen at any index, and
it is not the responsiblity of the downstream code to be aware of the
webspace. It may be that exceptions are thrown as a side-effect of a
bad configuration in a part of the system which has no knowledge of the
webspace.

Most errors would be caused by a bad webspace configuration during the
execution of this code and ensuring that the webspace path is available
to the developer in addition to the underlying exception will save a lot
of time and prevent the requirement of debugging (breakpoints, die()) to
find out which webspace caused the problem.

On Thu, Apr 14, 2016 at 12:00:43AM -0700, Daniel Rotter wrote:
>    In [1]src/Sulu/Component/Webspace/Manager/WebspaceCollectionBuilder.php:
> 
>  > -                $this->logger->warning(
>  > -                    sprintf(
>  > -                        'Error: "%s" in "%s". The file has been skipped.',
>  > -                        $e->getMessage(),
>  > -                        $file->getRealPath()
>  > -                    )
>  > -                );
>  > -            } catch (WebspaceLocalizationNotUsedException $e) {
>  > -                $this->logger->warning(
>  > -                    sprintf(
>  > -                        'Error: "%s" in "%s". The file has been skipped.',
>  > -                        $e->getMessage(),
>  > -                        $file->getRealPath()
>  > -                    )
>  > -                );
>  > +            } catch (\Exception $e) {
> 
>    If it is not a webspace related error the user also does not have to know
>    the path. And in all webspace related errors a reference to the webspace
>    is given, so it is quite easy to know the path.
> 
>    Symfony does, but it is very easily overseen. Same for PHPUnit.
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu/pull/2279#discussion_r59670627
>    2. https://github.com/sulu/sulu/pull/2279/files/488904213abfd5de2073dc30473f858099ce741a#r59670627
>    3. https://github.com/sulu/sulu/pull/2279/files/488904213abfd5de2073dc30473f858099ce741a#r59670627